File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String str) {
        Stack<Character> s=new Stack<>();
        for(int i=0;i<str.length();i++)
        {
            char ch=str.charAt(i);
            if(ch=='(' || ch=='{' || ch=='[')
                s.push(ch);
            else if(ch==')' && !s.isEmpty() && s.peek()=='(')
                s.pop();
            else if(ch=='}' && !s.isEmpty() && s.peek()=='{')
                s.pop();
            else if(ch==']' && !s.isEmpty() && s.peek()=='[')
                s.pop();
            else
                return false;
        }
        return s.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()
        brackets = {
            ')': '(',
            '}': '{',
            ']': '['
        }
        for ch in s:
            if ch in brackets.values():
                stack.append(ch)
            elif ch in brackets.keys():
                if stack:
                    last = stack.pop()
                    if brackets[ch] != last:
                        return False
                else:
                    return False

        if not stack:
            return True
        else:
            return False